base/server/python/pki/server/deployment/scriptlets/selinux_setup.py

```diff
@@ -168,41 +176,56 @@ def destroy(self, deployer):
-                        fcon.delete(
-                            deployer.mdict['pki_server_database_path'] +
-                            self.suffix, "")
+                        file_records = fcon.get_all()
+
+                        if self.is_exist(file_records,
+                                         deployer.mdict['pki_instance_path'] +
+                                         self.suffix):
+                            logger.info(
+                                "deleting selinux fcontext \"%s\"",
+                                deployer.mdict['pki_instance_path'] + self.suffix)
+                            fcon.delete(
+                                deployer.mdict['pki_instance_path'] +
+                                self.suffix, "")
+
+                        if self.is_exist(file_records,
+                                         deployer.mdict['pki_instance_log_path'] +
+                                         self.suffix):
+                            logger.info(
+                                "deleting selinux fcontext \"%s\"",
+                                deployer.mdict['pki_instance_log_path'] +
+                                self.suffix)
+                            fcon.delete(
+                                deployer.mdict['pki_instance_log_path'] +
+                                self.suffix, "")
+
+                        if self.is_exist(file_records,
+                                         deployer.mdict['pki_instance_configuration_path'] +
+                                         self.suffix):
+                            logger.info(
+                                "deleting selinux fcontext \"%s\"",
+                                deployer.mdict['pki_instance_configuration_path'] +
+                                self.suffix)
+                            fcon.delete(
+                                deployer.mdict['pki_instance_configuration_path'] +
+                                self.suffix, "")
+
+                        if self.is_exist(file_records,
+                                         deployer.mdict['pki_server_database_path'] +
+                                         self.suffix):
+                            logger.info(
+                                "deleting selinux fcontext \"%s\"",
+                                deployer.mdict['pki_server_database_path'] + self.suffix)
+                            fcon.delete(
+                                deployer.mdict['pki_server_database_path'] +
+                                self.suffix, "")
 
                         port_records = seobject.portRecords(trans)
+                        port_record_values = port_records.get_all()
                         for port in ports:
-                            logger.info("deleting selinux port %s", port)
-                            port_records.delete(port, "tcp")
+                            if self.is_exist(port_record_values, port):
```

Comment:
Yes, we should check first before deleting the port since deleting something that doesn't exist might fail the entire transaction.